### PR TITLE
dockerfiles for python

### PR DESCRIPTION
**Pull request by leewyang**
- `Dockerfile.python` - uses conda and cuda 11.5 (based on existing Scala dockerfile)
- `Dockerfile.pip` -uses pip and cuda 11.8 (based on Erik's init scripts for Databricks)

### CODE REVIEW

**Comment by lijinf2** [comment]
Looks good!

File: docker/Dockerfile.pip [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#ARG RAPIDS_ML_VER=spark-cuml
+#RUN git clone -b branch-$RAPIDS_ML_VER https://github.com/NVIDIA/spark-rapids-ml.git
+COPY . /spark-rapids-ml
+WORKDIR /spark-rapids-ml
+
+# install spark-rapids-ml
+RUN pip install --no-cache -r requirements_dev.txt \
+    && pip install --no-cache -e .
+
+SHELL ["conda", "run", "--no-capture-output", "-n", "base", "/bin/bash", "-c"]
```
Comment (eordentlich) [inline docker/Dockerfile.pip:58]: Do we still need Conda here?

File: docker/Dockerfile.python [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+###
```
Comment (eordentlich) [inline docker/Dockerfile.python:18]: Why retain all the commented out commands?

File: docker/Dockerfile.python [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#        --build-arg CUDA_VERSION=11.5.2 \
+#        --build-arg GCC_VERSION=9 \
+#        --build-arg CMAKE_VERSION=3.23.3 \
+#        --build-arg NINJA_VERSION=1.10.2 \
+#        --build-arg MAVEN_VERSION=3.8.6
+#
+###
+
+ARG CUDA_VERSION=11.5.2
+FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04
```
Comment (eordentlich) [inline docker/Dockerfile.python:39]: Same comment about devel base as for other file.

File: docker/Dockerfile.pip [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# install pyspark
+RUN pip install --no-cache-dir pyspark==${PYSPARK_VERSION}
+
+# Config JAVA_HOME
+ENV JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk-amd64
+
+#ARG RAPIDS_ML_VER=spark-cuml
+#RUN git clone -b branch-$RAPIDS_ML_VER https://github.com/NVIDIA/spark-rapids-ml.git
+COPY . /spark-rapids-ml
```
Comment (eordentlich) [inline docker/Dockerfile.pip:51]: Are we sure docker build will be run in src directory?

**Review by eordentlich — COMMENTED** [review]
We should alsoadd instructions for use to the docker dir README.md
https://github.com/NVIDIA/spark-rapids-ml/blob/spark-cuml/docker/README.md

File: docker/Dockerfile.pip [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ARG CUDA_VERSION=11.8.0
+FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04
```
[the remark this answers is not in the document]
Comment (leewyang) [reply docker/Dockerfile.pip:18]: Original docker image used `devel`, so I just kept it.  Just tried using a `runtime` image, but got errors during tests, so we can revisit slimming the image later.

File: docker/Dockerfile.pip (hunk 1 above)
Comment (leewyang) [reply docker/Dockerfile.pip]: Good catch... removed.

File: docker/Dockerfile.python (hunk 2 above)
Comment (leewyang) [reply docker/Dockerfile.python]: Vestige of original dockerfile... removed.

File: docker/Dockerfile.python (hunk 3 above)
Comment (leewyang) [reply docker/Dockerfile.python:18]: See reply above.

File: docker/Dockerfile.pip (hunk 4 above)
Comment (leewyang) [reply docker/Dockerfile.pip:51]: This is actually copying the entire repo (at the root-level) into the docker image, so the CI commands would look like:
```
git clone git@github.com:NVIDIA/spark-rapids-ml.git
cd spark-rapids-ml
docker build -t <image_tag> -f docker/Dockerfile .
```

**Review by eordentlich — APPROVED** [review]
👍   Should follow up with addition to README

**Pull request merged**